test/evp_test.c

```diff
@@ -3029,6 +3048,14 @@ static int parse(EVP_TEST *t)
     }
 
     for (pp++, i = 1; i < t->s.numpairs; pp++, i++) {
+        if (strcmp(pp->key, "Availablein") == 0) {
+            if (!prov_available(pp->value)) {
+                TEST_info("skipping, providers not available: %s:%d",
+                          t->s.test_file, t->s.start);
```

Comment:
If the legacy provider has been built then we should always run these tests with the legacy provider available.

Reply:
In fact we may want to run this twice. Once with default and legacy available, and once with fips available (but no default or legacy).

Reply:
Yes.  My thought is that this would be controlled with a config file.

Reply:
> Yes. My thought is that this would be controlled with a config file.

If so then that needs to be in scope of this PR. Otherwise we are losing a chunk of the tests with this PR.

Reply:
> If so then that needs to be in scope of this PR.

Autoloading of config files is part of another PR.  However, I can certainly add the necessary lines in `test/recipes/30-test_evp.t` to do this, even though currently useless (because autoloading isn't in place).  That will start working as soon as the autoloading PR is merged.
Is that what you're asking for?

Reply:
If we are relying on another PR for this to work then this one shouldn't be merged until that one is. Please add the necessary lines.

Reply:
Will do

Reply:
Done